File: api/src/main/kotlin/com/proyecto404/finalProjectJP/core/useCases/Followers.kt
package com.proyecto404.finalProjectJP.core.useCases

import com.proyecto404.finalProjectJP.core.domain.repositories.Relationships
import com.proyecto404.finalProjectJP.core.domain.repositories.Users
import com.proyecto404.finalProjectJP.core.domain.services.AuthService
import com.proyecto404.finalProjectJP.core.domain.services.SessionToken

class Followers(private val users: Users, private val relationships: Relationships, private val authService: AuthService) {
    fun exec(request: Request): Response {
        val followers = relationships.getFollowers(request.fromUser)
        authService.validateToken(request.token, users.get(request.requester))
        return Response(followers)
    }
    data class Request(val requester: String, val fromUser: String, val token: SessionToken)
    data class Response(val followers: List<String>)
}

File: web/src/core/useCases/Followers.ts
import {FollowingService} from "@/core/model/FollowingService";

export class Followers {
    constructor(private followingService: FollowingService) { }

    async exec(requester: string, fromUser: string, token: string) {
        return await this.followingService.getFollowers(requester, fromUser, token)
    }
}